web/elm/src/Application/Application.elm

```diff
@@ -426,20 +399,26 @@ urlUpdate : Routes.Route -> Model -> ( Model, List Effect )
 urlUpdate route model =
     let
         ( newSubmodel, subEffects ) =
-            if route == model.route then
+            if route == model.session.route then
                 ( model.subModel, [] )
 
             else if routeMatchesModel route model then
                 SubPage.urlUpdate
-                    { from = model.route
+                    { from = model.session.route
                     , to = route
                     }
                     ( model.subModel, [] )
 
             else
                 SubPage.init model.session route
+
+        oldSession =
+            model.session
+
+        newSession =
+            { oldSession | route = route }
```

Comment:
I **think** this is the place where we need to clear out the `.hovered` state

Now that the HoverState is tracked in the model, we'll need to clear it every time we navigate to a new page. Otherwise if you hover over a a job in regular pipeline card, clicking into it, and then navigating back to the dashboard page, you'll get a dangling hover text:

![Screen Shot 2020-11-20 at 4 26 01 PM](https://user-images.githubusercontent.com/17577073/99851467-39f89280-2b4d-11eb-9de2-c2e5ea8dfd94.png)
![Screen Shot 2020-11-20 at 4 26 11 PM](https://user-images.githubusercontent.com/17577073/99851469-3a912900-2b4d-11eb-9d0c-ffca61154171.png)

Reply:
Also, is it just me or is the `no instance vars` text slightly higher than the `foo:..` text?

Reply:
+1 on the line height (or padding/margin?) for `no instance vars` above. Nice catch, @chenbh !

Reply:
Two good catches, thanks! Fixed